pkg/apis/v1beta1/nodeclaim_validation.go

```diff
@@ -124,9 +124,6 @@ func (in *NodeClaimSpec) validateTaintsField(taints []v1.Taint, existing map[tai
 // NodeClaim requirements only support well known labels.
 func (in *NodeClaimSpec) validateRequirements() (errs *apis.FieldError) {
 	for i, requirement := range in.Requirements {
-		if requirement.Key == NodePoolLabelKey {
-			errs = errs.Also(apis.ErrInvalidArrayValue(fmt.Sprintf("%s is restricted", requirement.Key), "requirements", i))
-		}
```

Comment:
Should we be removing this? Are users now allowed to say that they want their pods to use a certain nodepool?

Reply:
Responded offline. We can't validate this on NodeClaims since NodeClaims will _always_ be assigned to a NodePool when they are being provisioned. It also makes sense to have a NodePool requirement on NodeClaims whereas it makes no sense to have this on NodePools because the worst you could do there is shoot yourself in the foot.